iceoryx_utils/include/iceoryx_utils/cxx/string.hpp

```diff
@@ -14,13 +14,61 @@
 #ifndef IOX_UTILS_CXX_STRING_HPP
 #define IOX_UTILS_CXX_STRING_HPP
 
+#include "iceoryx_utils/internal/cxx/string_internal.hpp"
+#include "optional.hpp"
+
 #include <cstring>
 #include <iostream>
 
 namespace iox
 {
 namespace cxx
 {
+/// @brief concatenates two fixed strings/string literals
+///
+/// @param [in] fixed strings/string literals to concatenate
+///
+/// @return a new fixed string with capacity equal to the sum of the capacities of the concatenated strings
+///
+/// @code
+///     string<5> fuu("cdefg");
+///     auto bar = iox::cxx::concatenate(fuu, "ahc");
+/// @endcode
+template <typename T1, typename T2>
+typename std::enable_if<(internal::IsCharArray<T1>::value || internal::IsCxxString<T1>::value)
+                            && (internal::IsCharArray<T2>::value || internal::IsCxxString<T2>::value),
+                        string<internal::GetCapa<T1>::capa + internal::GetCapa<T2>::capa>>::type
+concatenate(const T1& t1, const T2& t2);
+
+/// @brief concatenates an arbitrary number of fixed strings or string literals
+///
+/// @param [in] fixed strings/string literals to concatenate
+///
+/// @return a new fixed string with capacity equal to the sum of the capacities of the concatenated strings
+///
+/// @code
+///     string<4> fuu("cdef");
+///     auto bar = iox::cxx::concatenate(fuu, "g", "ah", fuu);
+/// @endcode
+template <typename T1, typename T2, typename... Targs>
+typename std::enable_if<(internal::IsCharArray<T1>::value || internal::IsCxxString<T1>::value)
+                            && (internal::IsCharArray<T2>::value || internal::IsCxxString<T2>::value),
+                        string<internal::SumCapa<T1, T2, Targs...>::value>>::type
```

Comment:
For loops in constexpr functions are a C++14 extensions as far as I know. But if we could use C++14 it would be a better way compile time calculating a sum.
The C++17 way is even better. With parameter unpacking we can write something like:
```cpp
template<int... N>
constexpr int sum() {
    return (N + ...);
}
```